processor/k8sattributesprocessor/options.go

```diff
@@ -125,6 +125,9 @@ func enabledAttributes() (attributes []string) {
 // If no fields explicitly provided, the defaults are pulled from metadata.yaml.
 func withExtractMetadata(fields ...string) option {
 	return func(p *kubernetesprocessor) error {
+		if fields == nil {
+			return nil
+		}
 		if len(fields) == 0 {
 			fields = enabledAttributes()
 		}
```

Comment:
@fatsheep9146 given that your fix in core is merged, we can move this part to `createDefaultConfig` and don't do anything else. Users will be able to set `metadata: []` which I believe is more correct way than setting `null`, but `null` will also be supported

Reply:
That make senses, I will modify like that.

Reply:
After modify like this, user should set `[]`, `null` is not supported, since the null should override the default values.

Reply:
you can check this logic here

https://github.com/open-telemetry/opentelemetry-collector/blob/e22d721b8b6867183d9d2404b5442fab0b2daa1e/confmap/confmap_test.go#L462-L471

@dmitryax

Reply:
I test this locally and make sure, for now only setting `metadata` to `[]` works, setting to `null` will keep the default values

Reply:
> I test this locally and make sure, for now only setting metadata to [] works, setting to null will keep the default values

This behavior makes total sense. We don't need to support `null` additionally for this